explorer/interpreter/type_checker.cpp

```diff
@@ -4588,6 +4588,20 @@ auto TypeChecker::DeclareClassDeclaration(Nonnull<ClassDeclaration*> class_decl,
       ScopeInfo::ForClassScope(scope_info, &class_scope, std::move(bindings));
   for (Nonnull<Declaration*> m : class_decl->members()) {
     CARBON_RETURN_IF_ERROR(DeclareDeclaration(m, class_scope_info));
+
+    const auto* fun = dyn_cast<FunctionDeclaration>(m);
+    if (!fun) {
+      continue;
+    }
+
+    if (fun->virt_override() == VirtualOverride::Impl) {
```

Comment:
```suggestion
    if (fun && fun->virt_override() == VirtualOverride::Impl) {
```
I think this is clearer than using `continue`, because it would make sense to add additional checks to this loop for other kinds of members.